ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,280 @@
+    for param_check in checks:
+        if isinstance(param_check, BaseInputArgumentChecker):
+            param_check.check()
+
+
+class BaseInputArgumentChecker(ABC):
+    """Abstract class to check input arguments"""
+
+    @abstractmethod
+    def check(self):
+        """Abstract method to check input arguments"""
+        raise NotImplementedError("The check is not implemented")
+
+
+class RequiredParamTypeCheck(BaseInputArgumentChecker):
+    """Class to check required input parameters"""
+
+    def __init__(self, parameter, parameter_name, expected_type):
+        self.parameter = parameter
+        self.parameter_name = parameter_name
+        self.expected_type = expected_type
+
+    def check(self):
+        """Method raises ValueError exception if required parameter has unexpected type"""
+        check_parameter_type(
+            parameter=self.parameter,
+            parameter_name=self.parameter_name,
+            expected_type=self.expected_type,
+        )
+
+
+class OptionalParamTypeCheck(RequiredParamTypeCheck):
+    """Class to check optional input parameters"""
+
+    def check(self):
+        """Method checks if optional parameter exists and raises ValueError exception if it has unexpected type"""
+        if self.parameter is not None:
+            check_parameter_type(
+                parameter=self.parameter,
+                parameter_name=self.parameter_name,
+                expected_type=self.expected_type,
+            )
+
+
+def check_file_extension(
+    file_path: str, file_path_name: str, expected_extensions: list
+):
+    """Function raises ValueError exception if file has unexpected extension"""
+    file_extension = file_path.split(".")[-1].lower()
+    if file_extension not in expected_extensions:
+        raise ValueError(
+            f"Unexpected extension of {file_path_name} file. expected: {expected_extensions} actual: {file_extension}"
+        )
+
+
+def check_that_null_character_absents_in_string(parameter: str, parameter_name: str):
+    """Function raises ValueError exception if null character: '\0' is specified in path to file"""
+    if "\0" in parameter:
+        raise ValueError(f"\\0 is specified in {parameter_name}: {parameter}")
+
+
+def check_that_file_exists(file_path: str, file_path_name: str):
+    """Function raises ValueError exception if file not exists"""
+    if not exists(file_path):
+        raise ValueError(
+            f"File {file_path} specified in '{file_path_name}' parameter not exists"
+        )
+
+
+def check_that_parameter_is_not_empty(parameter, parameter_name):
+    """Function raises ValueError if parameter is empty"""
+    if not parameter:
+        raise ValueError(f"parameter {parameter_name} is empty")
+
+
+def check_that_all_characters_printable(parameter, parameter_name):
+    """Function raises ValueError if one of string-parameter characters is not printable"""
+    if not all(c.isprintable() for c in parameter):
```

Comment:
Please, do something like that:
```suggestion
def check_that_all_characters_printable(parameter, parameter_name, allow_crlf=False):
    """Function raises ValueError if one of string-parameter characters is not printable"""
    if not allow_crlf:
        is_ok = all(c.isprintable() for c in parameter)
    else:
        is_ok = all((c.isprintable() or c=="\n" or c == "\r") for c in parameter)
    if not is_ok:
```